### ISSUE TITLE
Auto zooming

### ISSUE THREAD

**Issue by qlesz** [mikepauer/Carbonite#670]
With last commit for some reason when i change the zone (ie flying in to suramar city from broken shore) my map is automatically zooming in to some crazy factors (like instance zoom). So basically everytime im flying from one map to another i have to zoom out manually. My auto scale option is unchecked (off). Also when i switch between full map and minimised map it zooming in aswell if i hoover mouse over any other zone. Ok after some investigation i found "save map scale" option help with this issue but still after every zone change carb map is going back automatically to saved scale even if previous zone i just left was zoomed in or zoomed out.

**Comment by IrcDirk** [comment]
Which zone to which... becouse at my side all works ok.

**Comment by qlesz** [comment]
Nope. Still broken. here is link what exacly going on https://drive.google.com/open?id=0B5GoUBF5o1KEbml6SjFZRWQ0MU0

**Comment by IrcDirk** [comment]
U are sure u did updated it? It was before latest commit, now works fine.

**Comment by qlesz** [comment]
Ok to make sure i'm using latest version i've downloaded it again. It look like i've made mistake BUT i'm not sure it's intended to work this way,with latest version (and im sure its latest this time) it's still zooming in/out of previous scale to certain set point when i change the zone (regardless of saved map scale) but at least that point is not so annoying. So basically what changed from last time is zoom point map is automatically zooming in/out to is less annoying and "save map scale" does nothing now.

**Comment by IrcDirk** [comment]
Works with no problem to me, same as before changes i made. 

Use "save map scale" ten after auto zoom use "restore previous scale", this should make it stay the way u set it.

**Comment by qlesz** [comment]
Look like your advise was working till i get to instance. After i left instance my map get crazy again. So look like i need to do "restore map scale" after every instance.

**Comment by IrcDirk** [comment]
Yes, this is still an issue im working on it.

**Comment by IrcDirk** [comment]
Ok. Fixed but this is testing commit for restoring map scale after exiting Instance. Try now.

**Comment by qlesz** [comment]
Seem to be working ok now. I'll close it soon if nothing new won't come up.

**Comment by Palonblade** [comment]
Having issues with latest build zooming all the way in sometimes when logging in and changing zones. I have to constantly manually zoom back out. I usually have auto zoom unchecked.

**Comment by IrcDirk** [comment]
Describe those issues. Some previous comit did fucked up some scale saving so u may have problems with this. So as i wrote to @qlesz before: set desired scale then use "save map scale" and then use "restore map scale" (u can do it just after saving). This should make map not scale randomly.

**Comment by Palonblade** [comment]
Any zones, seems completely random at this point. I will try your suggestion.

**Comment by qlesz** [comment]
Look like you need to restore map scale on each character (or at least each server i didn't really paid attention on that) separatelly. That happened to me after last commit resolving this issue when i relogged on other toon where on my main was working fine already. I'll check on other toon and confirm is it each character or server.

**Comment by qlesz** [comment]
Yep "save map scale" "restore map scale" working fine. Just don't forget to restore it again if you relog on different server toon ;)

**Comment by qlesz** [comment]
Look like i closed it too fast. OK here what exacly happened. I've logged in on my alt i was playing after carb issue fix, so "save/restore" steps were done on that alt. When i logged in i was in dalaran. I ported to SW and map get zoomed in to minimum factor so i did "restore map scale" again.Then i went to Tol Barad instance to finish some legion skinning quest. When i ported back to dalaran map was zoomed in again.

**Comment by IrcDirk** [comment]
U sure u got latest? Just tested like 4 instances in and on out (leaving by foot and port back too) all zooms out with no problems.

**Comment by IrcDirk** [comment]
Pushed fix. Should work now.

**Comment by Palonblade** [comment]
Still having issues

**Comment by Palonblade** [comment]
Is there any setting I can change to possibly prevent this. Happens when logging into IF on my druid, flying from Dalaran to Broken Shore, Logging into or going to DK and Pally Order Hall, Sometimes when just simply flying across a zone and other random times it seems.

**Comment by Kuewal** [comment]
I am also having this issue, when I fly from zone to zone, my map zooms in
all the way

**Comment by IrcDirk** [comment]
Always use latest from here. Try to set desired scale then use "save map scale" and then use "restore map scale".

**Comment by Palonblade** [comment]
I have the latest and have saved and restored map scale. It does not prevent my minimap from auto zooming and I do not have auto scale set.  Without auto scale set why would the minimap zoom by itself at all?

**Comment by IrcDirk** [comment]
Specify those cases. Need as much details as possible: first zone, next zone which zooms in. Writing that this is random will not help.

**Comment by Palonblade** [comment]
I did in a previous post 2 days ago.

**Comment by IrcDirk** [comment]
Those maps are smaller or grater than main zone maps so if u zoom in in dalaran becouse u need to see details, then flying out of this zone map zoom will stay that way, so its zoomed in more than u are using it between normal zones in whitch u dont need to change map scale to see details.

**Comment by IrcDirk** [comment]
Other thing is that "Auto scale" is not for that what u think. It auto scales map to "go to" location or when u click watch button (circle beside objective name) in Carb quest watch window.

**Comment by IrcDirk** [comment]
Made some changes. @Palonblade test this. Go to Dalaran and "save map scale" to desired map scale, go to Broken Shore. map scale should not change, go back to Dalaran, map scale should not change. 

There is exemption to this: entering instance map will center map to this instance (so scale will autochange), exiting will restore map scale to saved using "save map scale".

Tell me if that makes a difference.

**Comment by Palonblade** [comment]
After todays commits:
This is Dalaran before Flying to Broken Shore after saving map scale:
http://tinypic.com/r/2vv73ox/9
This is Broken Shore after flying from Dalaran:
http://tinypic.com/r/mbtrms/9
This is when logging into my DK Order Hall:
http://tinypic.com/r/nmg9a9/9

Just a few examples. I didn't include any flying from zone to zone but can if needed.

**Comment by IrcDirk** [comment]
https://youtu.be/OkVfTw1mWmE

**Comment by Palonblade** [comment]
I do not have the Minimaps combined if that makes a difference

**Comment by IrcDirk** [comment]
Just checked, no difference with minimap not combined...

**Comment by IrcDirk** [comment]
And as i see in your screen 2, map scale did not changed in Broken Shore after leaving Dalaran...

**Comment by IrcDirk** [comment]
In DK did u moved? It should center and restore map scale.

**Comment by Palonblade** [comment]
At logon:
![wowscrnshot_060117_001239](https://cloud.githubusercontent.com/assets/22230769/26664383/37ffae86-465f-11e7-8c9f-d1938e1a7753.jpg)

Moved to Order Hall Board:
![wowscrnshot_060117_001310](https://cloud.githubusercontent.com/assets/22230769/26664388/44535bb0-465f-11e7-9758-8fdb74ec8b2b.jpg)

**Comment by IrcDirk** [comment]
Leave order halls by teleporting to dala then go back. Sometimes u need to do "restore map scale" once on every char, but do it only when teleporting and going back will not fix zoom.

On my pally and dk i dont have those issues.

**Comment by IrcDirk** [comment]
Oh and one thing to Pala order halls, this is overlay map indoor so it docks 100% size of zone. Dalaran  is map in map and its smaller than zone map. So map doesnt auto zoom and if u are in Dalaran and u zoom in to get details on map then go to Pala order halls, scale stays same, but map is big (100% size o fzone) so u have impression that scale changes. Check it your self using maximized map.

**Comment by Palonblade** [comment]
What do I need to do on my Pally and DK to prevent this. I've gone to Dalaran on my Pally and saved/restored map scale. Returned to my order Hall and it was not zoomed, so far so good. I exited the game and came back later. When logging into my Pally in the Order Hall it was zoomed again. 
![wowscrnshot_060117_173254](https://cloud.githubusercontent.com/assets/22230769/26709375/53a2cf4e-4720-11e7-8917-4ad32241ebb2.jpg)
![wowscrnshot_060117_231228](https://cloud.githubusercontent.com/assets/22230769/26709379/56383384-4720-11e7-8584-e9932d68963b.jpg)

**Comment by IrcDirk** [comment]
Cant reproduce. Map scale stays same after logout and login.

Did u do something between? Other char?

**Comment by Palonblade** [comment]
I did log into other characters.

**Comment by IrcDirk** [comment]
I did reloged too.

**Comment by IrcDirk** [comment]
Ok, it seems that Carbonite uses one profile for every character on realm. So when u zoom in on one character the zoom will be same on others. So when o reloged to other char and changed the zoom it sstayed the same on DK.

**Comment by IrcDirk** [comment]
Im working on fixing profiles. Then u will be able to have different setting on different chars.

**Comment by IrcDirk** [comment]
Ok. Profiles are fixed now. But u need create new profile for every char (or change to character specyfic profile). This will reset settings but u can copy them from "Default" global profile. 

After this all setting should be char specyfic. That way zoom settings will be saved for each char separately.
